ygen/protogen.go

```diff
@@ -594,3 +668,57 @@ func genListKeyProto(list *yangDirectory, listName string, listPackage string, s
 
 	return km, nil
 }
+
+// enumInProtoUnionField parses an enum that is within a union and returns the generated
+// enumeration that should be included within a protobuf message for it.
+func enumInProtoUnionField(name string, types []*yang.YangType) (map[string]*protoMsgEnum, error) {
+	enums := map[string]*protoMsgEnum{}
+	for _, t := range types {
+		if t.Kind == yang.Yenum && t.Name == "enumeration" {
+			n := fmt.Sprintf("%s", yang.CamelCase(name))
+			enum, err := genProtoEnum(&yang.Entry{
+				Name: n,
+				Type: t,
+			})
+			if err != nil {
+				return nil, err
+			}
+			enums[n] = enum
+		}
+	}
+
+	return enums, nil
+}
+
+// unionFieldToOneOf takes an input name, a yang.Entry containing a field definition and a mappedType
+// containing the proto type that the entry has been mapped to, and returns the list of fields within
+// the oneof that should be created, along with any enumerations that are contained within the union.
+func unionFieldToOneOf(name string, e *yang.Entry, mtype mappedType) ([]protoMsgField, map[string]*protoMsgEnum, error) {
+	enums, err := enumInProtoUnionField(name, e.Type.Type)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	typeNames := []string{}
+	for tn := range mtype.unionTypes {
+		typeNames = append(typeNames, tn)
+	}
+	sort.Strings(typeNames)
+
+	oofs := []protoMsgField{}
+	for _, t := range typeNames {
+		// Split the type name on "." to ensure that we don't have oneof options
+		// that reference some other package in the type name.
+		tp := strings.Split(t, ".")
+		tn := tp[len(tp)-1]
+		st := protoMsgField{
+			Name: fmt.Sprintf("%s_%s", name, tn),
+			Type: t,
+			// TODO(robjs): Determine how best to output tags here.
+			Tag: 42,
```

Comment:
Worth mentioning here or elsewhere that this is dependent on some other external feature implementation (maybe open an issue?) Then you can decide whether to implement a short term solution to make code functional or wait for longer term solution.

Reply:
Updated the comment, there's a nuance here that I'll explain in person.